shared/src/types/api/question.dto.ts

```diff
@@ -0,0 +1,16 @@
+import { DifficultyLevel, Language } from "../base";
+
+export class GetQuestionReq {
+  difficultyLevel: DifficultyLevel;
+}
+
+export class GetQuestionRes {
+  id: number;
+  title: string;
+  //   category: string;
+  //   topics: string[];
+  description: string;
+  hints: string[];
+  templates: { language: Language; code: string }[];
+  testcases: { inputs: string[]; output: string }[];
```

Comment:
```suggestion
```
The test cases and templates will probably be loaded by the other services so we don't have to return them in this endpoint.